### ISSUE TITLE
Consuming private typescript modules

### ISSUE THREAD

**Issue by jacobdr** [meteor-typescript/meteor-typescript-compiler#53]
Our team is running into build errors when trying to consume (i.e. import ) raw `.ts` files from a package in the `node_modules` directory into our project. 

We are trying to consume a package that is solely responsible for logging and that is installed via `git+ssh` via `meteor npm install`. That same package is consumed in an identical way in other pure-Typescript, non-Meteor projects without issue. However, I think that during the meteor link/build process the dependency is not being bundled. I don't see the package included in the `.meteor/local/build/programs/server/packages` path like my other `package.json`/npm dependencies.

In my Meteor file I have the following source that tries to perform the import:
```
import * as  HistoryLoggerReference from 'bellona-history/src/HistoryLogger';
console.log("DEBUG9", HistoryLoggerReference);
```

Meteor throws the following error: 
```
Error: Can't find npm module 'bellona-history/src/HistoryLogger'. Did you forget to call 'Npm.depends' in package.js within the 'modules-runtime' package?
    at Object.Npm.require (/Users/Jacob/Documents/Projects/CDL/bellona/bellona-ui/src/.meteor/local/build/programs/server/boot.js:195:17)
    at options.fallback (packages/modules-runtime/modules-runtime.js:21:1)
    at require (packages/modules-runtime/.npm/package/node_modules/install/install.js:81:1)
    at meteorInstall.server.history.historyInstance.ts (server/history/historyInstance.ts:1:1)
    at fileEvaluate (packages/modules-runtime/.npm/package/node_modules/install/install.js:141:1)
    at require (packages/modules-runtime/.npm/package/node_modules/install/install.js:75:1)
    at GlobalCollections.ts:17:3
    at /Users/Jacob/Documents/Projects/CDL/bellona/bellona-ui/src/.meteor/local/build/programs/server/boot.js:283:10
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/Jacob/.meteor/packages/meteor-tool/.1.3.2_4.108qz0f++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/underscore/underscore.js:79:11)
    at /Users/Jacob/Documents/Projects/CDL/bellona/bellona-ui/src/.meteor/local/build/programs/server/boot.js:133:5
```

I should mention that the project compiles fine when I just run the `tsc` compiler, which further leads me to believe this is related to the Meteor build process not bundling the dependency. 

Has anybody seen this issue before or have advice as to how I might be able to get things to work?

**Comment by barbatus** [comment]
Is `bellona-history` an NPM? If yes have you installed it? Also what does it mean to consume raw `.ts`?
NPM packages should be based on JavaScript files.

**Comment by jacobdr** [comment]
`bellona-history` is the name of a private package that we host on bitbucket, and install via git+ssh. It is installed and present in the local `node_modules` directory, and present in package.json. 

We don't compile the source `.ts` files in the `bellona-history` repository before publishing -- we essentially are using npm to ship (and version) our `.ts` files, which are consumed directly in Meteor. 

So locally in my project I have a path `node_modules/bellona-history/src/HistoryLogger`. However, when Meteor builds, there is no `bellona-history` directory in the built `packages` folder.

**Comment by barbatus** [comment]
I don't think it's right to ship `.ts` as with NPM, NPM is designed for JS modules. If you want to bring `.ts` files to a meteor project and compile them at place, use an Atmosphere package.

**Comment by jacobdr** [comment]
Thanks for your quick replies so far. Do you have any examples of the latter approach?

**Comment by barbatus** [comment]
Actually, it's not good as well. Why do you need to ship `.ts` instead of compiled JS modules as most do BTW?

**Comment by jacobdr** [comment]
I suppose its not a requirement but its a pattern that we follow in all of our non-Meteor Typescript projects, and works really well. No need to mess around with git hooks, compile and declare typings in `package.json` or `typings.json`, etc. You just get everything you need as a Typescript consumer by making sure that the file can be resolved in the `tsconfig`. 

Do you have any speculation as to why I am running into this issue? I'm pretty confident it has to do with the special way that Meteor handles processing of `node_modules`... I would contribute a fix or at the least documentation, but I am pretty perplexed as to where to start.

Any suggestions/thoughts, Mr. TS guru?

**Comment by jacobdr** [comment]
For future reference to others who come across this issue, the solution I ended up following:

1. Made typescript and typings a dependency of my private package (`npm install --save typescript typings`)
2. Added an "install" lifecycle hook the the package.json so that it reads:

   ```
"scripts": {
        "install": "typings install && tsc -d"
}
   ```  
3. On the consumer side in my Meteor project I import from the `outDir` specified in my dependency's `package.json` (my naming convention is a little odd since I didn't change the rootDir, so it looks something like `import * as  HistoryLoggerReference from 'bellona-history/dist/src/HistoryLogger';`)
